### ISSUE TITLE
Exporting a function named `html` provokes a 404 (`index.html.html`)

### ISSUE THREAD

**Issue by JulianCataldo** [TypeStrong/typedoc#2497]
Hello,

It looks like doing this:

```js
// NOTE: Just `export { html } from '@lit-labs/ssr';` will not show in TypeDoc.
import { html as litHtml } from '@lit-labs/ssr';
export const html = litHtml;
```

is producing this page : `index.html.html`.

That will trigger a 404 page.

Renaming to `htmll`, for example, solve the problem.

**Comment by JulianCataldo** [comment]
Just an additional note, when deploying to GH Pages, it does work.
It might be correlated to trailing slash behavior.

**Comment by Gerrit0** [comment]
I can't reproduce this. I copied your code into a new project, installed the latest TypeDoc version + `@lit-labs/ssr`, and ran TypeDoc on that file. The generated docs appear to work just fine, producing this:

```
~/Desktop/td_tests> tree docs
docs
├── assets
│   ├── highlight.css
│   ├── main.js
│   ├── navigation.js
│   ├── search.js
│   └── style.css
├── functions
│   └── html.html
├── index.html
└── modules.html
```

**Comment by JulianCataldo** [comment]
Hello. Indeed it's working in production, when it's served from an env which add a trailing slash (Vercel, GH Pages…).
In dev. it broke, _I think_, because of that and maybe another factor.
You're right, I'm gonna put up a minimal repro. and get back to you.
Thanks!

**Comment by Gerrit0** [comment]
Looking at that, `index.html.html` is the expected filename, as the member is named `html` and is in a module named `index`. Links to it look like they work...

**Comment by JulianCataldo** [comment]
Oh, I've see I've overlooked an important fact: in local dev, I use `npx serve docs`.

Only with `index.html.html`, I've got the 404 + redirect to `functions`. Trailing slash has nothing to do with it, as I suspected at first.

<img width="404" alt="Screenshot 2024-02-18 at 20 45 16" src="https://github.com/TypeStrong/typedoc/assets/603498/299f5173-ba84-454b-9639-33601b48d23c">

Maybe overly zealous Regex?

So, I conclude that this is a [`serve`](https://github.com/vercel/serve/issues) issue, not a Typedoc one!

Thank you. Closing
